psycop/projects/sequence_models/datasets.py

```diff
@@ -0,0 +1,54 @@
+"""
+datasets for training and evaluation of sequence models
+"""
+
+import datetime as dt
+
+from psycop.common.feature_generation.loaders.raw.load_ids import SplitName
+from psycop.common.feature_generation.sequences.cohort_definer_to_prediction_times import \
+    CohortToPredictionTimes
+from psycop.common.feature_generation.sequences.patient_loaders import (
+    DiagnosisLoader, PatientLoader)
+from psycop.common.sequence_models.dataset import (PatientSliceDataset,
+                                                   PatientSlicesWithLabels)
+from psycop.common.sequence_models.registry import Registry
+from psycop.projects.t2d.feature_generation.cohort_definition.t2d_cohort_definer import \
+    T2DCohortDefiner
+
+
+@Registry.datasets.register("patient_slice_dataset")
+def create_patient_slice_dataset(
+    min_n_visits: int,
+    split_name: str,  # TODO: @martin hvordan ville du type hinte det her?
```

Comment:
@MartinBernstorff is there a way to type hint this better?

Reply:
Literal["Name1", "Name2"] etc. 

Personligt ville jeg bruge en enum så jeg kunne bruge et match-statement til exhaustiveness-checks, men vil forvente at confection/pydantic ikke kan re-caste fra strengen i .cfg-filen til enum'en :-)

Reply:
Kunne godt være den kunne, men antager den ikke kan. Den bliver recasted som noget af det første dog.

Reply:
Yeah, lad os gå med `Literal`